src/Canvas.tsx

```diff
@@ -264,19 +272,55 @@ const InternalCanvas: FC<CanvasProps & { ref?: Ref<CanvasRef> }> = forwardRef(
     }));
 
     const mount = useRef<boolean>(false);
+    const dragNodeData = useMemo(
+      () => getDragNodeData(canvasDragNode, layout?.children),
+      [canvasDragNode, layout?.children]
+    );
+    const [dragNodeDataWithChildren, setDragNodeDataWithChildren] = useState<{
+      [key: string]: any;
+    }>(dragNodeData);
     useLayoutEffect(() => {
       if (!mount.current && layout !== null && xy[0] > 0 && xy[1] > 0) {
         mount.current = true;
       }
     }, [layout, xy]);
 
-    const dragNodeData = useMemo(
-      () =>
-        rest.dragNode
-          ? layout?.children?.find((c) => c.id === rest.dragNode.id)
-          : null,
-      [layout?.children, rest.dragNode]
-    );
+    const createDragNodeChildren = (children: any) => {
+      if (!children || !Array.isArray(children)) {
+        return [];
+      }
+
+      return children.map(({ children, ...n }) => {
+        const element =
+          typeof dragNode === 'function' ? dragNode(n as NodeProps) : dragNode;
+        return (
+          <CloneElement<NodeProps>
+            key={`${id}-node-${n.id}-node-drag`}
+            element={element}
+            disabled
+            children={element.props.children}
+            animated={animated}
+            nodes={children}
+            childEdge={edge}
+            childNode={dragNode}
+            {...n}
+            onDragStart={event => {
+              setDragType(event.dragType);
+            }}
+            id={`${id}-node-${n.id}-node-drag`}
+          />
+        );
+      });
+    };
+
+    useEffect(() => {
+      if (dragNodeData.children) {
+        const nodeCopy = { ...dragNodeData };
+        // Node children is expecting a list of React Elements, need to create a list of elements
+        nodeCopy.children = createDragNodeChildren(nodeCopy.children);
+        setDragNodeDataWithChildren(nodeCopy);
+      }
+    }, [createDragNodeChildren, dragNodeData]);
```

Comment:
You are gonna need to add `useCallback` back - the effect will always thrash now :(